### ISSUE TITLE
Issue: Overlap Between Training and Test Sets

### ISSUE THREAD

**Issue by Hhankyangg** [WisconsinAIVision/YoLLaVA#4]
While training tokens for my new concept using `train-multi-token.py`, I noticed that the `image_paths` in the test set are identical to those in the training set. This could potentially lead to data leakage and unreliable evaluation metrics.

### Details

When constructing the test set, I observed the following code snippet:

```python
test_dataset = PersonalizedDataset(
    data_root=args.data_root,
    sks_name=args.sks_name,
    train_image_paths=train_dataset.images_path,  # This seems unusual
    tokenizer=tokenizer,
    config=model.config,
    image_processor=image_processor,
    device=model.device,
    set='test',
)
```

It appears that the test set is using the same `train_image_paths` as the training set, which suggests that the images in the test set are not being differentiated from those in the training set.

Additionally, after checking the `PersonalizedDataset` class in `my_llava.py`, I found the following:

```python
for x in np.unique(train_image_paths):
    self.image_paths.append(x)
```

From this, it seems that the test set is constructed using the same image paths as the training set, without any changes in the paths during test data retrieval. This suggests an issue in the way the test set is built, resulting in an overlap between the training and test datasets.

### Verification

To confirm this, I printed out the `image_paths` for both the training and test sets, and they were indeed identical. This overlap could undermine the evaluation process and model performance metrics. It would be better if the test set used distinct images not seen during training.

**Comment by thaoshibe** [comment]
Hello, thank you for your concern.
In the training code, I **purposely** set the train/set image path to be the same, as I want to save the checkpoint that have the **best accuracy** on the train set!

https://github.com/WisconsinAIVision/YoLLaVA/blob/9810b53cc8e714de74d706c2565b74067ed435f9/train.py#L246

The evaluation code is different and will not use this test_dataset ^^.
But I agree with you, name the variable like that cause confusion for reader~ I'll edit it soon!

**Comment by thaoshibe** [comment]
I do it as it would be actually unfair to save the best checkpoint that _perform the best on the test-set while training_.
In training, the test-set should **not** be seen.

Hope this clarify your concern~ Thank you ;)
(I'll close this issues as it seems a misunderstanding. Feel free to re-open or raise any other concerns you might have!)
